File: Code_Vita2018/square_free.py
from math import sqrt

def isSquareFree(n):
    if n % 2 == 0:
        n = n / 2
    if n % 2 == 0:
        return False
    for i in range(3, int(sqrt(n) + 1)):
        if n % i == 0:
            n = n / i
        if n % i == 0:
            return False
    return True

def ans_val(num):
	count=0
	for i in range(2,num+1):
		if num%i==0:
			if isSquareFree(i):
				count+=1
	return count

value=int(input())
ans=ans_val(value)
print(ans,end='')

File: Code_Vita2018/square_free.cpp
#include<bits/stdc++.h>
using namespace std;
bool isSquareFree(int n)
{
    if (n % 2 == 0)
       n = n/2;
    if (n % 2 == 0)
       return false;
    for (int i = 3; i <= sqrt(n); i = i+2)
    {
        if (n % i == 0)
        {
           n = n/i;
           if (n % i == 0)
               return false;
        }
    }
 
    return true;
}
std::vector<int> printDivisors(int n)
{
    std::vector<int> v;
    for (int i=2; i<=sqrt(n); i++)
    {
        if (n%i == 0)
        {
            if (n/i == i)
                v.push_back(i);
            else 
                {v.push_back(i);
            	v.push_back(n/i);}
        }
    }
    return v;
}
int main(){
	int n;int count=0;
	cin>>n;
	std::vector<int> v=printDivisors(n);
	for(int i=0;i<v.size();i++){
		if(isSquareFree(v[i])){
			count+=1;
		}
	}
	cout<<count<<"\n";
}